### ISSUE TITLE
Nested decorators with ParamSpec no longer support args and kwargs attributes

### ISSUE THREAD

**Issue by Rapptz** [microsoft/pyright#3327]
**Describe the bug**

In the newest release, nested decorators no longer support `P.args` and `P.kwargs`.

**To Reproduce**
Have a decorator that returns a callable that wraps another callable and have it use `ParamSpec` attributes.

Running pyright on this results in an error:

```
bug.py
  bug.py:23:26 - error: Argument of type "P.args" cannot be assigned to parameter of type "P@deprecated"
    Type "P.args" cannot be assigned to type "P@deprecated" (reportGeneralTypeIssues)
  bug.py:23:34 - error: Argument of type "P.kwargs" cannot be assigned to parameter of type "P@deprecated"
    Type "P.kwargs" cannot be assigned to type "P@deprecated" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations
```

**Expected behavior**
No error.

**Screenshots or Code**
```py
from __future__ import annotations

import warnings
import functools
from typing import Optional, Callable, TypeVar
from typing_extensions import ParamSpec

T = TypeVar('T')
P = ParamSpec('P')

def deprecated(instead: Optional[str] = None) -> Callable[[Callable[P, T]], Callable[P, T]]:
    def actual_decorator(func: Callable[P, T]) -> Callable[P, T]:
        @functools.wraps(func)
        def decorated(*args: P.args, **kwargs: P.kwargs) -> T:
            warnings.simplefilter('always', DeprecationWarning)  # turn off filter
            if instead:
                fmt = "{0.__name__} is deprecated, use {1} instead."
            else:
                fmt = '{0.__name__} is deprecated.'

            warnings.warn(fmt.format(func, instead), stacklevel=3, category=DeprecationWarning)
            warnings.simplefilter('default', DeprecationWarning)  # reset filter
            return func(*args, **kwargs)

        return decorated

    return actual_decorator
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

Command line

```
> pyright --version
pyright 1.1.236
```

**Additional context**

This works in mypy and the previous version of pyright (1.1.235).

**Comment by erictraut** [comment]
Apologies for the regression. I've fixed the problem and included a new unit test based on your code sample to prevent similar regressions in the future. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.237, which I just published. It will also be included in the next release of pylance.
